src/net.rs

```diff
@@ -177,6 +180,57 @@ pub trait HostResponse {
     fn finish(&mut self) -> Result<(), Error>;
 }
 
+/// Represents a physical port that can be used to interact with client devices.
+///
+/// This for example is used for a PA RoT to send messages to a AC RoT.
+///
+/// This trait provides a generic mechanism for sending and receiving requests
+/// from devices.
+pub trait DevicePort<'a> {
+    /// The send function takes a `header` and `msg` to send to a device.
+    /// The implementation of this function must send the header and msg
+    /// to the device specified by `dest`.
+    ///
+    /// This should block until the operation is complete.
+    ///
+    /// On success it should return the number of bytes sent.
+    /// On a failure an error is returned.
+    fn send(
+        &mut self,
+        dest: u8,
+        header: Header,
+        msg: &'a [u8],
+    ) -> Result<usize, Error>;
+
+    /// This is a blocking call that should wait for a response or a timeout.
+    ///
+    /// `timeout`: Number of milliseconds to wait.
```

Comment:
Can this be a `Duration` instead?

Also, hmm, I wonder if the contract should specify that the timeout is best effort... it might be the case that the underlying hardware just doesn't support a timeout. Thoughts?